go-controller/pkg/ovn/namespace.go

```diff
@@ -409,21 +410,44 @@ func (oc *Controller) deleteNamespaceLocked(ns string) *namespaceInfo {
 }
 
 func (oc *Controller) createNamespaceAddrSetAllPods(ns string) (addressset.AddressSet, error) {
-	// Get all the pods in the namespace and append their IP to the address_set
 	var ips []net.IP
-	existingPods, err := oc.watchFactory.GetPods(ns)
-	if err != nil {
-		klog.Errorf("Failed to get all the pods (%v)", err)
-	} else {
-		ips = make([]net.IP, 0, len(existingPods))
-		for _, pod := range existingPods {
-			if pod.Status.PodIP != "" && !pod.Spec.HostNetwork {
-				podIPs, err := util.GetAllPodIPs(pod)
+	if config.Kubernetes.HostNetworkNamespace != "" &&
+		ns == config.Kubernetes.HostNetworkNamespace {
+		// add the mp0 interface addresses to this namespace.
+		existingNodes, err := oc.watchFactory.GetNodes()
+		if err != nil {
+			klog.Errorf("Failed to get all nodes (%v)", err)
+		} else {
+			ips = make([]net.IP, 0, len(existingNodes))
+			for _, node := range existingNodes {
+				hostSubnets, err := util.ParseNodeHostSubnetAnnotation(node)
 				if err != nil {
-					klog.Warningf(err.Error())
+					klog.Errorf("Error parsing host subnet annotation for node %s (%v)",
+						node.Name, err)
 					continue
 				}
-				ips = append(ips, podIPs...)
+				for _, hostSubnet := range hostSubnets {
+					mgmtIfAddr := util.GetNodeManagementIfAddr(hostSubnet)
+					ips = append(ips, mgmtIfAddr.IP)
+				}
+			}
+		}
+	} else {
```

Comment:
I'm not sure we should skip the standard processing; it feels to me like the feature is less invasive if it adds some weird functionality on top but doesn't affect the normal functionality...